src/main/webapp/app/exercise/import/from-file/exercise-import-from-file.component.spec.ts

```diff
@@ -198,6 +199,78 @@ describe('ExerciseImportFromFileComponent', () => {
         // THEN
         expect(component.fileForImport).toEqual(file);
     });
+
+    // Verifies that exercise categories are correctly parsed from stringified JSON entries during import.
+    it('should correctly parse exercise categories during import', async () => {
+        // GIVEN
+        component.exerciseType = ExerciseType.PROGRAMMING;
+
+        const progEx = new ProgrammingExercise(undefined, undefined);
+        progEx.id = 999;
+        progEx.title = 'Category test exercise';
+        progEx.type = ExerciseType.PROGRAMMING;
+
+        (progEx as any).categories = ['{"color":"#0d3cc2","category":"Testing categories"}', '{"color":"#691b0b","category":"Issue"}'];
+
+        const zip = new JSZip();
+        zip.file('exercise.json', JSON.stringify(progEx));
+        const zipBlob = await zip.generateAsync({ type: 'blob' });
+
+        component.fileForImport = zipBlob as any;
+        const openImportSpy = jest.spyOn(component, 'openImport');
+        await fixture.whenStable();
+
+        // WHEN
+        await component.uploadExercise();
+        await fixture.whenStable();
+
+        // THEN
+        const importedEx = component.exercise as ProgrammingExercise;
+        expect(importedEx).toBeDefined();
+        expect(importedEx.categories!).toHaveLength(2);
+
+        expect(importedEx.categories![0]).toBeInstanceOf(ExerciseCategory);
+        expect(importedEx.categories![0].category).toBe('Testing categories');
+        expect(importedEx.categories![0].color).toBe('#0d3cc2');
+
+        expect(importedEx.categories![1]).toBeInstanceOf(ExerciseCategory);
+        expect(importedEx.categories![1].category).toBe('Issue');
+        expect(importedEx.categories![1].color).toBe('#691b0b');
+
+        expect(openImportSpy).toHaveBeenCalledOnce();
+        expect(openImportSpy).toHaveBeenCalledWith(importedEx);
+    });
+
+    // Ensures backward compatibility when old exercise JSONs do not contain 'categories' field.
+    // parseExerciseCategories() should safely handle undefined categories without throwing errors.
+    it('should import exercise correctly even if categories are missing', async () => {
+        // GIVEN
+        component.exerciseType = ExerciseType.PROGRAMMING;
+        const progEx = new ProgrammingExercise(undefined, undefined);
+        progEx.id = 555;
+        progEx.title = 'No category exercise';
+        progEx.type = ExerciseType.PROGRAMMING;
+
+        delete (progEx as any).categories;
+
+        const zip = new JSZip();
+        zip.file('exercise.json', JSON.stringify(progEx));
+        const zipBlob = await zip.generateAsync({ type: 'blob' });
+
+        component.fileForImport = zipBlob as any;
+        const openImportSpy = jest.spyOn(component, 'openImport');
+
```

Comment:
Why is `await fixture.whenStable();` in comparison to "should correctly parse exercise categories during import" missing here?